docs/solidity-by-example.rst

```diff
@@ -645,99 +645,71 @@ Safe Remote Purchase
+       (explained later), and the mechanism for sending it does not matter.
     4. Bob claims their payment by presenting the signed message to the smart contract, it verifies the
        authenticity of the message and then releases the funds.
 
 Creating the signature
 ----------------------
 
-Alice does not need to interact with Ethereum network to
-sign the transaction, the proccess is completely offline.
-In this tutorial, we will sign messages in the browser
-using ``web3.js`` and ``MetaMask``.
-In particular, we will use the standard way described in `EIP-762 <https://github.com/ethereum/EIPs/pull/712>`_,
+Alice does not need to interact with the Ethereum network to sign the transaction, the process is completely offline.
+In this tutorial, we will sign messages in the browser using `web3.js <https://github.com/ethereum/web3.js>`_ and `MetaMask <https://metamask.io>`_, using the method described in `EIP-762 <https://github.com/ethereum/EIPs/pull/712>`_,
 as it provides a number of other security benefits.
 
 ::
 
-    /// Hashing first makes a few things easier
-    var hash = web3.sha3("message to sign");
-    web3.personal.sign(hash, web3.eth.defaultAccount, function () {...});
-
+    /// Hashing first makes things easier
+    var hash = web3.utils.sha3("message to sign");
+    web3.eth.personal.sign(hash, web3.eth.defaultAccount, function () {
+      console.log("Signed")
+    });
 
-Note that the ``web3.personal.sign`` prepends the length of the message to the signed data.
-Since we hash first, the message will always be exactly 32 bytes long,
-and thus this length prefix is always the same, making everything easier.
+.. note::
+  The ``web3.eth.personal.sign`` prepends the length of the message to the signed data. Since we hash first, the message will always be exactly 32 bytes long, and thus this length prefix is always the same.
 
 What to Sign
 ------------
 
-For a contract that fulfills payments, the signed message must include:
+For a contract that fulfils payments, the signed message must include:
 
-    1. The recipient's address
-    2. The amount to be transferred
-    3. Protection against replay attacks
+    1. The recipient's address.
+    2. The amount to be transferred.
+    3. Protection against replay attacks.
 
-A replay attack is when a signed message is reused to claim authorization for
-a second action.
-To avoid replay attacks we will use the same as in Ethereum transactions
-themselves, a so-called nonce, which is the number of transactions sent by an 
-account.
-The smart contract will check if a nonce is used multiple times.
+A replay attack is when a signed message is reused to claim authorisation for a second action. To avoid replay attacks we will use the same technique as in Ethereum transactions themselves, 'nonce', which is the number of transactions sent by an account. The smart contract will check if a nonce is used multiple times.
```

Comment:
`a so-called "nonce"` sounds better to me.

Reply:
@chriseth not too keen on that phrase, try this as an alternative.